### PR TITLE
fix: add actions index options and adapt actions for pymeilisearch

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**

### CODE REVIEW

File: doc-deploy-index/action.yml [hunk 1]
```diff
@@ -75,6 +75,14 @@ inputs:
     default: false
     type: string
 
+  pymeilisearchopts:
+    description: >
+      Set of options to pass to `pymeilisearch <https://pymeilisearch.docs.ansys.com/version/stable/>`_
+      while scraping.Default is None.
```
Comment (RobPasMue) [inline doc-deploy-index/action.yml:81]:
```suggestion
      while scraping. Default is ``None``.
```

File: .github/workflows/ci_cd.yml [hunk 2]
```diff
@@ -238,3 +254,36 @@ jobs:
...
+    needs: doc-deploy-stable
+    steps:
+      - name: "Install Git and clone project"
+        uses: actions/checkout@v3
+
+      - name: "get version of library"
+        shell: bash
+        run: |
+          version=$(cat VERSION)
+          echo "ver=$version" >> $GITHUB_ENV
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:272]:
```suggestion
          echo "FULL_VERSION=$(cat VERSION)" >> $GITHUB_ENV
```

File: .github/workflows/ci_cd.yml [hunk 3]
```diff
@@ -238,3 +254,36 @@ jobs:
...
+
+      - name: Scrape the stable documentation to PyMeilisearch
+        run: |
+          VERSION=$(python -c "import os; version=os.environ['ver']; print('.'.join(version.split('.')[:2]))")
+          VERSION_MEILI=$(python -c "import os; version=os.environ['ver']; print('-'.join(version.split('.')[:2]))")
```
Comment (RobPasMue) [inline .github/workflows/ci_cd.yml:277]:
```suggestion
          VERSION=$(python -c "import os; version=os.environ['FULL_VERSION']; print('.'.join(version.split('.')[:2]))")
          VERSION_MEILI=$(python -c "import os; version=os.environ['FULL_VERSION']; print('-'.join(version.split('.')[:2]))")
```

**Review by RobPasMue — APPROVED** [review]
Some suggestions - otherwise LGTM

File: .github/workflows/ci_cd.yml [hunk 4]
```diff
@@ -186,6 +188,20 @@ jobs:
...
+  doc-index-dev:
+    name: "Deploy dev index docs"
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: doc-deploy-dev
+    steps:
+      - name: "Deploy the latest documentation index"
+        uses: ansys/actions/doc-deploy-index@v4.1
+        with:
+          cname: "$DOCUMENTATION_CNAME/version/dev"
```
[the remark this answers is not in the document]
Comment (Revathyvenugopal162) [reply .github/workflows/ci_cd.yml:200]: no we are scrapping the local pages using localhost, it is not url scraping

**Comment by Revathyvenugopal162** [comment]
Thank you so much for review @RobPasMue , will merge after setting the env variable.

File: .github/workflows/ci_cd.yml [hunk 5]
```diff
@@ -12,6 +12,8 @@ env:
   MAIN_PYTHON_VERSION: '3.10'
   DOCUMENTATION_CNAME: 'actions.docs.ansys.com'
   test-library-name: 'ansys-actions'
+  MEILISEARCH_API_KEY: ${{ secrets.MEILISEARCH_API_KEY }}
+  MEILISEARCH_HOST_URL: ${{ secrets.MEILISEARCH_HOST_UR }}
```
Comment (Revathyvenugopal162) [inline .github/workflows/ci_cd.yml:16]:
```suggestion
```

**Pull request merged**